File: LeetCode/153. Find Minimum in Rotated Sorted Array.cpp
// Author : @AdarshGaur

class Solution {
public:
    int findMin(vector<int>& nums) {
        int lo = 0, hi = (int)nums.size() -1, min;
        while(lo <= hi){
            int mid = lo + (hi-lo)/2;
            if(nums[mid] <= nums.back()){
                hi = mid -1;
                min = nums[mid];
            }else{
                lo = mid +1;
            }
        }
        return min;
    }
};

File: LeetCode/153. Find Minimum in Rotated Sorted Array.java
class Solution {
    public int findMin(int[] nums) {
        int leftIdx = 0, rightIdx = nums.length -1;
        int midIdx = (leftIdx + rightIdx)>>1;
        while(leftIdx < rightIdx){
            if(nums[midIdx] >= nums[rightIdx]){
                leftIdx = midIdx +1;
            }else{
                rightIdx = midIdx;
            }
            midIdx = (leftIdx + rightIdx)/2;
        }
        return nums[leftIdx];
    }
}

